src/api/entities/Instruction/index.ts

```diff
@@ -175,6 +178,44 @@ export class Instruction extends Entity<UniqueIdentifiers, string> {
     return statusResult === InternalInstructionStatus.Pending;
   }
 
+  /**
+   * Retrieve current status of the Instruction. This can be subscribed to know if instruction fails
+   *
+   * @note can be subscribed to
+   * @note current status as `Executed` means that the Instruction has been executed/rejected and pruned from
+   *   the chain.
+   */
+  public async onStatusChange(callback: SubCallback<InstructionStatus>): Promise<UnsubCallback> {
+    const {
+      context: {
+        polymeshApi: {
+          query: {
+            settlement: { instructionDetails },
+          },
+        },
+      },
+      id,
+      context,
+    } = this;
+
+    const assembleResult = (rawStatus: MeshInstructionStatus): InstructionStatus => {
+      const status = meshInstructionStatusToInstructionStatus(rawStatus);
+
+      if (status === InternalInstructionStatus.Pending) {
+        return InstructionStatus.Pending;
+      } else if (status === InternalInstructionStatus.Failed) {
+        return InstructionStatus.Failed;
+      } else {
+        // TODO @prashantasdeveloper remove this once the chain handles Executed/Rejected state separately
+        return InstructionStatus.Executed;
```

Comment:
For now, I have added a comment and clarified in the docs about the status - Executed. Once the core team makes changes to handle the status separately, I think we can live with this

Reply:
I'd prefer to have a status for `ExecutedOrRejected` explaining we currently can't know which is the correct one. But assuming an instruction was executed when it was rejected could lead to other issues. 
Could you also ask Robert to check when is this feature coming to the chain?

Reply:
@VictorVicente Do you just mean change the status name for documentation purposes?

If the core team is going to add Rejected in the future, then I think its better to leave this as just `Executed`, otherwise when we separate them it will force a breaking change or else we'll be left with a name that's verbosely misleading.

We should add a TS Doc `@note` to the status explaining it though.

Reply:
On second though, adding in `Rejected` should still be a major version thing, since people should look at their code again, and if they do a `never` check it could break TS projects.

I am kind of liking `ExecutedOrRejected` now.

Reply:
This change will suppose a renaming change for current instruction status enum. We've decided to leave as it is, and to create a new one `Rejected` once it's provided directly from the chain.